### DISCUSSION TITLE
Make Sketchware Pro project more customizable

### DISCUSSION THREAD

**Discussion by oong819** [Sketchware-Pro/Sketchware-Pro#812] [Ideas]
# Problem
i think Sketchware Pro should be more customizable

the current version are annoying to user that import their own drawable, use their one Java/Kotlin code and install their own library

they might want some more controls over their project

# Solutions

1. chang image view image choosing from the current one to edittext (with button open the current one)

2. make all library appear in library tab (gson, bumptech glide and more auto added instead of tell user to turn it on in settings (like AppCompat)

3. seperate "hidden options", like if i lower the min SDK level below Android version support multidex, AndroidX multidec library auto added

**Comment by j-px** [comment]
You already have full control over projects. Use the inject property and add `tools:replace="android:src"` in there if you want to enter an image source manually.

To 2nd: More auto-added? Use Local libraries, if that's what you're talking about. Otherwise, the toast makes sense, that way users know that AppCompat is needed for more advanced components like MaterialButton.

To 3rd: MultiDex already gets added if minSdkVersion is under 21.

**Reply by oong819** [reply]
at 2 and 3 i mean the opposite, library are auto added without telling me nor have options

when i lower the sdk lv to below 21 MultiDex also auto added without telling me nor have options

sorry for my bad English skill

**Reply by j-px** [reply]
Alright, understood. Your suggestion is toggles to disable adding MultiDex, Gson and Glide to a project although it's currently possible by using Exclude built-in libraries.
